src/test/java/com/fasterxml/jackson/failing/JsonIdentityInfoAndBackReferences3964Test.java

```diff
@@ -0,0 +1,297 @@
+            return fruit;
+        }
+
+        public void setFruit(Fruit fruit) {
+            this.fruit = fruit;
+        }
+    }
+    
+    /**
+     * Fails : Lean version that fails and Without getters and setters
+     */
+    @JsonIdentityInfo(
+            generator = ObjectIdGenerators.PropertyGenerator.class,
+            property = "id",
+            scope = Animal.class
+    )
+    public static class Animal {
+        public final int id;
+        public List<Cat> cats;
+
+        @JsonCreator
+        public Animal(@JsonProperty("id") int id, @JsonProperty("cats") List<Cat> cats) {
+            this.id = id;
+            this.cats = cats;
+        }
+    }
+
+    @JsonIdentityInfo(
+            generator = ObjectIdGenerators.PropertyGenerator.class,
+            property = "id",
+            scope = Cat.class
+    )
+    public static class Cat {
+        public int id;
+        public List<Food> foods;
+        @JsonBackReference("id")
+        public Animal animal;
+
+        @JsonCreator
+        public Cat(@JsonProperty("id") int id, @JsonProperty("foods") List<Food> foods) {
+            this.id = id;
+            this.foods = foods;
+        }
+    }
+
+    @JsonIdentityInfo(
+            generator = ObjectIdGenerators.PropertyGenerator.class,
+            property = "id",
+            scope = Food.class
+    )
+    public static class Food {
+        public int id;
+        public Cat cat;
+
+        @JsonCreator
+        public Food(@JsonProperty("id") int id) {
+            this.id = id;
+        }
+    }
+
+    /**
+     * Passes : Testing lean without getters and setters
+     * and also without {@link JsonCreator}.
+     */
+    @JsonIdentityInfo(
+            generator = ObjectIdGenerators.PropertyGenerator.class,
+            property = "id",
+            scope = Fish.class
+    )
+    public static class Fish {
+        public int id;
+        public List<Squid> squids;
+    }
+
+    @JsonIdentityInfo(
+            generator = ObjectIdGenerators.PropertyGenerator.class,
+            property = "id",
+            scope = Squid.class
+    )
+    public static class Squid {
+        public int id;
+        public List<Shrimp> shrimps;
+        @JsonBackReference("id")
+        public Fish fish;
+    }
+
+    @JsonIdentityInfo(
+            generator = ObjectIdGenerators.PropertyGenerator.class,
+            property = "id",
+            scope = Shrimp.class
+    )
+    public static class Shrimp {
+        public int id;
+        public Squid squid;
+    }
+
+    /*
+    /**********************************************************************
+    /* Test methods
+    /**********************************************************************
+    */
+
+    final ObjectMapper MAPPER = jsonMapperBuilder()
+            .enable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES).build();
+
+    /**
+     * Fails : Original test
+     */
+    public void testOriginal() throws JsonProcessingException {
```

Comment:
Could you change this to just throw `Exception` (since 3.0 has different base exception)

Reply:
Sorry about this repeated forgetting to change to Exception repeatedly 🥲 

I will find a mechanim to prevent this from happening 👍🏻👍🏻